### PR TITLE
Add MicroG and reddit extended support

### PR DESCRIPTION
**Pull request by IMXEren**
This is proposed in response to add extended patch support for microg (any flavor: `TeamVanced/VancedMicroG`, `inotia00/VancedMicroG`, `inotia00/mMicroG`; only patched by extended) and reddit.

Kindly have a look at it.

### CODE REVIEW

File: src/downloader/utils.py [hunk 1]
```diff
@@ -29,10 +29,15 @@ def download_revanced(config: RevancedConfig, patcher: Patches) -> None:
             ["inotia00", "revanced-integrations", config.integrations_apk],
             ["inotia00", "revanced-patches", config.patches_jar],
         ]
-    if "youtube" in config.apps or "youtube_music" in config.apps:
-        assets += [
-            ["inotia00", "mMicroG", "mMicroG-output.apk"],
-        ]
+    if "youtube" in config.apps or "youtube_music" in config.apps or "MicroG" in config.apps:
```
Comment (nikhilbadyal) [inline src/downloader/utils.py:32]: Can you fix the lint [issue](https://results.pre-commit.ci/run/github/524139377/1689338087.Li2XkX_AQoa8zvuAZ8uYcg) by running.

`pip install pre-commit && pre-commit run --all-files`.

**Review by nikhilbadyal — CHANGES_REQUESTED** [review]

File: src/downloader/utils.py (hunk 1 above)
Comment (IMXEren) [reply src/downloader/utils.py]: Done!

**Comment by nikhilbadyal** [comment]
I have tested and there are few issues with the PR. It will fail when patching microg.

Even though the file was downloaded already [here](https://github.com/nikhilbadyal/docker-py-revanced/pull/218/files#diff-7addf02859a4b1133fb2055b4544497e9a0c9c186bd307c2c4c9965b9c0f6547R32). 

The builder isn't aware of that and will try to download it again [here](https://github.com/nikhilbadyal/docker-py-revanced/blob/a0617da9a820a10015c2790a06f4503454520f62/main.py#L34). When not mentioned the source to download it will try to download it from [APKMIRROR](https://github.com/nikhilbadyal/docker-py-revanced/blob/a0617da9a820a10015c2790a06f4503454520f62/src/downloader/factory.py#L35) causing it to fail.

**Comment by nikhilbadyal** [comment]
I have added support for both these in [this](https://github.com/nikhilbadyal/docker-py-revanced/pull/219) and [this](https://github.com/nikhilbadyal/docker-py-revanced/pull/220) PR.

**Comment by IMXEren** [comment]
I knew about that. I just didn't try to edit the `.env` file in the PR to add `EXISTING_DOWNLOADED_APKS=MicroG`.

Edit: I see you've added it to download it.

**Pull request closed without merging**